app/controllers/admin/submissions_controller.rb

```diff
@@ -1,11 +1,85 @@
 # frozen_string_literal: true
 
+class StateActions
```

Comment:
It would be nice if this was in its own file so that it's clearly discoverable via cmd+p and sidebar.

Reply:
You're totally right I was being lazy, haha. I extracted to a separate file with 3d1ceb7 and even added some specs as punishment. 😝